src/nat/middleware/function_middleware.py

```
@@ -0,0 +1,186 @@
+    async def middleware_invoke(self, value: Any, call_next: CallNext, context: FunctionMiddlewareContext) -> Any:
+        """Delegate to function_middleware_invoke for function-specific handling."""
+        return await self.function_middleware_invoke(value, call_next, context)
+
+    async def middleware_stream(self, value: Any, call_next: CallNextStream,
+                                context: FunctionMiddlewareContext) -> AsyncIterator[Any]:
+        """Delegate to function_middleware_stream for function-specific handling."""
+        async for chunk in self.function_middleware_stream(value, call_next, context):
+            yield chunk
+
+    async def function_middleware_invoke(self, value: Any, call_next: CallNext,
+                                         context: FunctionMiddlewareContext) -> Any:
```

Ruff (0.14.4)

Unused method argument: `context`

(ARG002)